app/models/cashbox.rb

```diff
@@ -0,0 +1,7 @@
+class Cashbox < ActiveRecord::Base
+  has_many :deposits, class_name: 'Payment', foreign_key: 'destination_id'
+  has_many :withdrawls, class_name: 'Payment', foreign_key: 'source_id'
```

Comment:
I agree, instead of Cashbox having two foreign ids maybe each Payment has two foreign ids-- source and destination